plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConfig.java

```diff
@@ -178,13 +179,29 @@ public int getMaxPartitionsPerWriter()
     }
 
     @Config("iceberg.max-partitions-per-writer")
-    @ConfigDescription("Maximum number of partitions per writer")
+    @ConfigDescription("Default maximum number of partitions handled per writer")
     public IcebergConfig setMaxPartitionsPerWriter(int maxPartitionsPerWriter)
     {
         this.maxPartitionsPerWriter = maxPartitionsPerWriter;
         return this;
     }
 
+    // If no upper limit is specified, the default max partitions per writer value is used which prevents users from
+    // submitting a session property value that increases the number of writers.
+    @Min(1)
+    public int getMaxPartitionsPerWriterLimit()
+    {
+        return maxPartitionsPerWriterLimit.orElse(maxPartitionsPerWriter);
+    }
+
+    @Config("iceberg.max-partitions-per-writer-limit")
```

Comment:
I don't like the addition of yet another property just to put a cap on the user session override.
Why can't we use the existing `iceberg.max-partitions-per-writer` as the cap on user override ?
I think we should follow the example of `query_max_memory_per_node` (and other similar query guard rails) and allow the session property to only be lesser than the config property.
Let the admin raise the default `iceberg.max-partitions-per-writer` if there is a need for users to exceed that, or just raise the default value for `iceberg.max-partitions-per-writer` in this PR if you think this is set too low today.

Reply:
Thanks for your feedback and we also thought of doing like this before. The problem is if we use the existing ```iceberg.max-partitions-per-writer``` as the cap, if we want to increase this cap for a single user, all the users that use this cluster will have a higher cap, effectively affecting the reliability of the cluster. So we introduce another config called ```iceberg.max-partitions-per-writer-limiy``` here, such that even if a specific user want to use a higher value of ```max-partitions-per-writer```, other users are not affected by this change and will continue to use the current ```iceberg.max-partitions-per-writer``` as the cap. That's why we implement it this way here. Please let me know if you have questions/feedback!

Reply:
Good point- I suppose you could use a session property manager config to set a default value lower than the maximum allowed limit and get the same result. @SongChujun - I think we can update the PR to remove the new configuration property